src/app/services/qr-verifier.service.ts

```diff
@@ -8,20 +8,42 @@ import { environment } from 'src/environments/environment';
 export class QrVerifierService {
 
   constructor() { }
-  pollServer(thePopup:Window, router:Router, state:string):void {
+  private _intervalId: number | undefined;
+  public get intervalId(): number | undefined {
+    return this._intervalId;
+  }
+  private set intervalId(value: number | undefined) {
+    this._intervalId = value;
+  }
+  private fetchServer(thePopup:Window, state:string):void {
           fetch(`${environment.BASE_URL}${environment.pollURL}?state=${state}`).then(
             (response) =>{
             
               if (response.status === 400 || response.status === 500) {
-                router.navigate(['/'])
+                window.location.replace('/')
```

Comment:
If there is an error I dont think you need to redirect the page anymore. This was done because of the QR window was shown in a different page